src/inscription.rs

```diff
@@ -99,19 +102,41 @@ impl<'a> InscriptionParser<'a> {
 
   fn parse_inscription(&mut self) -> Result<Option<Inscription>> {
     if self.advance()? == Instruction::Op(opcodes::all::OP_IF) {
-      let content = self.advance()?;
+      let media_type = self.advance()?;
 
-      let content = if let Instruction::PushBytes(bytes) = content {
+      let media_type = if let Instruction::PushBytes(bytes) = media_type {
         str::from_utf8(bytes).map_err(InscriptionError::Utf8Decode)?
       } else {
         return Err(InscriptionError::InvalidInscription);
       };
 
+      let inscription = match media_type {
+        "text/plain;charset=utf-8" => {
+          let content = if let Instruction::PushBytes(bytes) = self.advance()? {
```

Comment:
Move advance outside of the match, and convert to `Vec<u8>`. (There is a Vec<u8> -> String conversion you can use if it's text.